staticaddr/loopin/actions.go

```diff
@@ -118,21 +118,53 @@ func (f *FSM) InitHtlcAction(_ fsm.EventContext) fsm.EventType {
 	}
 
 	f.loopIn.HtlcCltvExpiry = loopInResp.HtlcExpiry
+
+	// Retrieve htlc tx server nonces.
 	f.htlcServerNonces, err = toNonces(loopInResp.HtlcServerNonces)
 	if err != nil {
 		return f.HandleError(err)
 	}
+	f.htlcServerNoncesHighFee, err = toNonces(
+		loopInResp.HtlcServerNoncesHighFee,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+	f.htlcServerNoncesExtremelyHighFee, err = toNonces(
+		loopInResp.HtlcServerNoncesExtremelyHighFee,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
 
 	// We need to defend against the server setting high fees for the htlc
 	// tx since we might have to sweep the timeout path. We maximally allow
 	// 20% of the swap value to be spent on fees.
-	f.loopIn.HtlcTxFeeRate = chainfee.SatPerKWeight(loopInResp.HtlcFeeRate)
-	fee := f.loopIn.HtlcTxFeeRate.FeeForWeight(f.loopIn.htlcWeight())
+	feeRate := chainfee.SatPerKWeight(loopInResp.HtlcFeeRate)
+	fee := feeRate.FeeForWeight(f.loopIn.htlcWeight())
+	if fee > f.loopIn.TotalDepositAmount()/5 {
```

Comment:
nit: instead of writing `TotalDepositAmount()/5` everywhere, please add a variable tracking this value.